cogs/user_commands.py

```diff
@@ -124,6 +124,82 @@ async def giveitem(self, ctx:utils.Context, user:discord.Member, *, item_type:st
 
         # And now we done
         return await ctx.send(f"{ctx.author.mention}, sent 1x **{self.bot.items[item_type.replace(' ', '_').lower()].display_name}** to {user.mention}!")
+    
+    @utils.command()
+    @commands.bot_has_permissions(send_messages=True,embed_links=True)
+    async def keys(self, ctx:utils.Context):
+        """
+        Check all users who have a key to your garden
+        """
+
+        async with self.bot.database() as db:
+            key_owners = await db("SELECT * FROM user_garden_access WHERE garden_owner=$1",ctx.author.id)
+
+            if not key_owners:
+                return await ctx.send(f"No one else has a key to your garden")
+            
+            embed = utils.Embed(use_random_colour=True,description=f"<@{ctx.author.id}>'s allowed users ({len(key_owners)})")
+            embed_fields = []
+            for key_owner in key_owners:
+                embed_fields.append(f"<@{key_owner['garden_access']}>")
+            
+            embed.add_field("Key owners:", '\n'.join(sorted(embed_fields)), inline=True)
+            
+            return await ctx.send(embed=embed)
+
+    @utils.command()
+    @commands.bot_has_permissions(send_messages=True)
+    async def givekey(self, ctx:utils.Context, user:discord.Member):
+        """
+        Give a key to your garden to another member.
+        """
+
+        if user.bot:
+            return await ctx.send("Bots can't help you maintain your garden.")
+        if user.id == ctx.author.id:
+            return await ctx.send("You already have a key.")
+
+        async with self.bot.database() as db:
+            given_key = await db("SELECT * FROM user_garden_access WHERE garden_owner=$1 AND garden_access=$2", ctx.author.id, user.id)
+            if given_key:
+                return await ctx.send(f"They already have a key!")
+            
+            await db(
+                "INSERT INTO user_garden_access (garden_owner, garden_access) VALUES ($1, $2)",
+                ctx.author.id, user.id
+            )
+            return await ctx.send(f"Gave {user.mention} a key!")
+
+    @utils.command()
+    @commands.bot_has_permissions(send_messages=True)
+    async def revokekey(self, ctx:utils.Context, user:typing.Optional[discord.Member], username:typing.Optional[str]):
```

Comment:
It's an interesting question. Personally I think it's important for usability by Discord noobs, someone sees a user in their "keys" list that they no longer want there, but they're no longer on a server with them. They'll be confused when the keys list correctly mentions them, but because their mention fails it doesn't work.

Alternatively, we could index the keys list, allowing someone to do `p.revokekeys 1` to revoke the first person on that list, and give a hint at the end of the embed about it.